README.md

```diff
@@ -88,13 +89,71 @@ Additionally, for the development we use:
 
 > See the [Development](#development) section for details on how to install `mockery`.
 
-# Installation
+# Quick-Start Installation
 
-## From binaries
+We provide an installation script to automatically install and update the latest version of Trento.
+
+## Trento Server Installation
+
+T.B.D.
+
+## Trento Agent Installation
+
+You can `curl | bash` if you want to live on the edge.
+
+```
+$ curl -sfL https://raw.githubusercontent.com/trento-project/trento/main/install.sh | sudo bash
+```
+
+Or you can fetch the script, and then execute it manually.
+
+```
+$ curl -O https://raw.githubusercontent.com/trento-project/trento/main/install.sh
+$ chmod 700 install.sh
+$ sudo ./install.sh
+```
+
+The script will ask you for two IP addresses.
+
+- `agent bind IP`: the private address to which the trento-agent should be bound for internal communications.
+  This is an IP address that should be reachable by the other nodes, including the trento server.
```

Comment:
Maybe "reachable from the other nodes". An exert required #@lee-martin .

Reply:
@aleksei-burlakov  @fabriziosestito 

I think I would propose something like the text below.  In general please avoid using the word "node", for Trento we use the word "hostname" to mean that, so we should be consistent.  Also, why should the Trento Agent be reachable by anything other than the Trento server? I'm not currently aware of any need for this.

"
- `agent bind IP`: the local IP address to which the trento-agent should bound to for Trento internal communications.
  This is IP address that must be reachable by the Trento server.
"

Reply:
> @aleksei-burlakov @fabriziosestito
> 
> I think I would propose something like the text below. In general please avoid using the word "node", for Trento we use the word "hostname" to mean that, so we should be consistent. Also, why should the Trento Agent be reachable by anything other than the Trento server? I'm not currently aware of any need for this.
> 
> "
> 
>     * `agent bind IP`: the local IP address to which the trento-agent should bound to for Trento internal communications.
>       This is IP address that must be reachable by the Trento server.
>       "

@lee-martin not sure we can change the word "node" with the word "hostname" in this context, they have a completely different meaning.  Maybe we could find another word for it, "host" maybe?

About the second point:
Looking at the consul architecture it was my understanding that the agents need to contact each others because of consul gossip protocol, see here: https://www.consul.io/docs/architecture and here: https://learn.hashicorp.com/tutorials/consul/reference-architecture?utm_source=consul.io&utm_medium=docs#network-connectivity-details

Reply:
"host" is the term to use instead of "node".
I agree that "hostname" doesn't fit here, as that's a property of a host.

Reply:
@fabriziosestito  Yes, absolutely "host" is what my brain was thinking, but my fingers automatically typed hostname. In trento we of course use the word "host" (without name) - my typo, sorry.

Reply:
> About the second point:
> Looking at the consul architecture it was my understanding that the agents need to contact each others because of consul gossip protocol, see here: https://www.consul.io/docs/architecture and here: https://learn.hashicorp.com/tutorials/consul/reference-architecture?utm_source=consul.io&utm_medium=docs#network-connectivity-details

@fabriziosestito  I understand your point from consul, but this has pretty major security implications for the actual SAP systems we will be managing. They should not all all be able to openly communicate with each other, e.g. prod and dev will frequently be separated. Ideally I would like to proceed that only the tento server speaks to the individual agents, @arbulu89 & @stefanotorresi  - do you see any issue with not allowing the agents to talk to each other? I would really like to avoid the intercommunication if possible.

Reply:
We do require intercommunication across all the Trento Agent peers, at the moment.
We might find other solutions to this, let's follow-up somewhere else.